x/metadata/keeper/scope.go

```diff
@@ -127,66 +129,109 @@ func (k Keeper) RemoveScope(ctx sdk.Context, id types.MetadataAddress) {
 
 	// Sessions will be removed as the last record in each is deleted.
 
-	k.clearScopeIndex(ctx, scope)
+	k.indexScope(ctx, nil, &scope)
 	store.Delete(id)
 	k.EmitEvent(ctx, types.NewEventScopeDeleted(scope.ScopeId))
 	defer types.GetIncObjFunc(types.TLType_Scope, types.TLAction_Deleted)
 }
 
-// clearScopeIndex delete any index records for this scope
-func (k Keeper) clearScopeIndex(ctx sdk.Context, scope types.Scope) {
-	store := ctx.KVStore(k.storeKey)
+// scopeIndexValues is a struct containing the values used to index a scope.
+type scopeIndexValues struct {
+	ScopeID         types.MetadataAddress
+	Addresses       []string
```

Comment:
curious why not typesafety for addresses and valueowner?

Reply:
I just used the same types that they are in the scopes. Same with the other similar `*IndexValues` structs.